test/index.test.js

```diff
@@ -911,5 +911,29 @@ describe('mongoose module:', function() {
       const m = new mongoose.Mongoose();
       assert.deepEqual(await m.syncIndexes(), {});
     });
+    it('Allows for the removal of indexes via string or object (gh-11547)', async function() {
+
+      const schema = new Schema({
+        title: String,
+        weight: Number,
+        age: Number,
+        name: String,
+        location: String
+      });
+
+      schema.index({ title: 1 });
+      schema.index({ weight: 1 });
+      schema.index({ age: 1, name: 1 });
+      schema.index({ location: 1 });
+      assert.equal(schema._indexes.length, 4);
+      const startingIndexes = JSON.parse(JSON.stringify(schema._indexes));
+      schema.removeIndex('title');
+      schema.removeIndex({ weight: 1 });
+      schema.removeIndex({ age: 1, name: 1 });
+      assert.equal(schema._indexes.length, 1);
+      assert.notDeepStrictEqual(startingIndexes, schema._indexes);
```

Comment:
Add an assert that checks that `schema._indexes[0]` is deep equal `{ location: 1 }`